homeassistant/components/hyperion/config_flow.py

```diff
@@ -0,0 +1,422 @@
+"""Hyperion config flow."""
+
+import asyncio
+import logging
+from typing import Any, Dict, Optional
+from urllib.parse import urlparse
+
+from hyperion import client, const
+import voluptuous as vol
+
+from homeassistant import config_entries
+from homeassistant.components.ssdp import ATTR_SSDP_LOCATION, ATTR_UPNP_SERIAL
+from homeassistant.const import CONF_BASE, CONF_HOST, CONF_ID, CONF_PORT, CONF_TOKEN
+from homeassistant.core import callback
+from homeassistant.helpers.typing import ConfigType
+
+# pylint: disable=unused-import
+from .const import (
+    CONF_AUTH_ID,
+    CONF_CREATE_TOKEN,
+    CONF_HYPERION_URL,
+    CONF_PRIORITY,
+    DEFAULT_ORIGIN,
+    DEFAULT_PRIORITY,
+    DOMAIN,
+)
+
+_LOGGER = logging.getLogger(__name__)
+_LOGGER.setLevel(logging.DEBUG)
+
+#     +-----------------------+
+#     |Step: SSDP             |
+# --->|                       |
+#     |Input: <discovery data>|
+#     +-----------------------+
+#           |
+#           |
+#           v
+#     +----------------+    +--------------------+
+#     |Step: user      |    |Step: import        |
+# --->|                |<---|                    |<---
+#     |Input: host/port|    |Input: <import data>|
+#     +----------------+    +--------------------+
+#           |
+#           |    Auth        +------------+
+#           |    required?   |Step: auth  |
+#           +--------------->|            |
+# Auth not  |                |Input: token|
+# required? |                +------------+
+#           |    Static         |
+#           v    token?         |
+#            <------------------+
+#           |                   |
+#           |                   |New token?
+#           |                   v
+#           |            +------------------+
+#           |            |Step: create_token|
+#           |            +------------------+
+#           |                   |
+#           |                   v
+#           |            +---------------------------+   +--------------------------------+
+#           |            |Step: create_token_external|-->|Step: create_token_external_fail|
+#           |            +---------------------------+   +--------------------------------+
+#           |                   |
+#           |                   v
+#           |            +-----------------------------------+
+#           |            |Step: create_token_external_success|
+#           |            +-----------------------------------+
+#           |                   |
+#           v                   |
+#     +----------------+        |
+#     | Step: Confirm  |<-------+
```

Comment:
1. User shouldn't need to confirm after already entering their host
2. The SSDP step should directly jump to auth or confirm step